scripts/gameBoard.js

```diff
@@ -1,92 +1,88 @@
-  var y = Math.floor((yClick-boarder)/squareSize)+1;
-  var xInSquare = ((x*(squareSize))+boarder-5) - xClick;
-  var yInSquare = ((y*(squareSize))+boarder-5) - yClick;
+  handleRandomClickedChip(x,y) {
+    var value = Math.floor(Math.random() * Math.floor(6)+1);
+    var colour = colours[Math.floor(Math.random() * Math.floor(4))];
 
-  // If clicked on the edges of the square, do NOT draw the Chip (square == [40, 40])
-  if(xInSquare <= 5 || xInSquare >= 35 || yInSquare <= 5 || yInSquare >= 35){
-    return null;
-  }
+    var boardSquare = this.board.find(square => {
+      if(square.xCoordinate === x && square.yCoordinate === y) return true;
+    });
 
-  return [x,y];
-}
+    if(boardSquare.bottomChip === null){
+      if(boardSquare.activeChip != null) {
+        boardSquare.bottomChip = boardSquare.activeChip;
+        draw.bottomChip(x,y, boardSquare.bottomChip.colour, 3);
+      }
 
-draw.getCanvas().addEventListener('click',e => {
-  xClick = e.clientX;
-  yClick = e.clientY;
-  var coordinates = getClickCoordinates(xClick,yClick);
-  if(coordinates) {
-    var x = coordinates[0];
-    var y = coordinates[1];
-    if (x>=0 && x<=gameBoardWidth && y>=0 && y<=gameBoardHeight) {
-      handleRandomClickedChip(x,y);
+      //TODO get rid of this from here
+      var chip = player.playChip(x, y, value);
+      draw.chip(chip);
+      boardSquare.activeChip = chip;
     }
+    console.log(boardSquare);
   }
-} , false);
 
-function handleRandomClickedChip(x,y) {
-  var value = Math.floor(Math.random() * Math.floor(6)+1);
-  var colour = colours[Math.floor(Math.random() * Math.floor(4))];
 
-  var boardSquare = board.find(square => {
-    if(square.xCoordinate === x && square.yCoordinate === y) return true;
-  });
+  getClickCoordinates(xClick,yClick) {
+    var border = gameBoardFrameSize*2;
+    var xLimit = (border+(squareSize*gameBoardWidth)-2);
+    var yLimit = (border+(squareSize*gameBoardHeight)-2);
 
-  if(boardSquare.bottomChip === null){
-    if(boardSquare.activeChip != null) {
-      boardSquare.bottomChip = boardSquare.activeChip;
-      draw.bottomChip(x,y, boardSquare.bottomChip.colour, 3);
+    if(xClick <= border || yClick <= border || xClick >= xLimit || yClick >= yLimit){
+      return null;
     }
 
-    //TODO get rid of this from here
-    var chip = player.playChip(x, y, value);
+    var x = Math.floor((xClick-border)/squareSize)+1;
+    var y = Math.floor((yClick-border)/squareSize)+1;
+    var xInSquare = ((x*(squareSize))+border-5) - xClick;
+    var yInSquare = ((y*(squareSize))+border-5) - yClick;
 
-    draw.chip(chip);
-    boardSquare.activeChip = chip;
+    // If clicked on the edges of the square, do NOT draw the Chip (square == [40, 40])
+    if(xInSquare <= 5 || xInSquare >= 35 || yInSquare <= 5 || yInSquare >= 35){
+      return null;
+    }
+
+    return [x,y];
   }
 
-  console.log(boardSquare);
+
+  reloadPage() {
+    window.location.reload(false);
+  }
```

Comment:
We could also remove this function from here, and the button from the index.html.
Otherwise, we should add a function to main which calls this function, so the button continues working.